File: weeks/02/0_lonely_integer/lonely_integer.py
def lonelyinteger(a):

    appears_once = set()
    appears_twice = set()
    
    for item in a:
        if item not in appears_once:
            appears_once.add( item )
        else:
            appears_twice.add( item )
        
    difference = appears_once-appears_twice
    lonely = list(difference)[0]
    return lonely

File: weeks/02/0_lonely_integer/lonely_integer.rb
def lonelyinteger(a)
    appears_once = Set.new
    appears_twice = Set.new
  
    a.each do |item|
      if !appears_once.include?(item)
        appears_once.add(item)
      else
        appears_twice.add(item)
      end
    end
  
    difference = appears_once - appears_twice
    
    difference.each do |lonely|
        return lonely
    end
  end
  
